### ISSUE TITLE
Support for Motion-Sensor, Door-Sensor and Smart Plug

### ISSUE THREAD

**Issue by sporca** [bramoosterhuis/com.devolo#6]
Do you plan to support further devices like the Devolo Motion (3-in-1) Sensor or the Door/Window (3-in-1) or the Smart Plug? I heard those are Philio Sensors with customized chip and firmware.

I managed to use the code from the philio app and change the hardware id to devolo. it works quite good, but sometimes the motion sensor hangs in alarm status, which causes a lot of problems.

EDIT: for me it was important to disable the illumination and temperatur differential report, because the sensor spammed their values to the homey and crashed the z-wave.

**Comment by bramoosterhuis** [comment]
Yesterday I added these devices, not sure if it all works correct, I don't own those devices. However,   you can install the alpha branch using the developer tools. If you miss some additional configuration please add it by forking and do a pull request, or just poke me :-).

**Comment by sporca** [comment]
Wow great news. Will try all the devices end of this week and report back.

**Comment by sporca** [comment]
started testing. found mistake in app.json. "productTypeId": is wrong for new devices.

I changed it for MT2647 to "2", for MT2646 to "1" and for MT2648 to "2". It then finds the correct app during inclusion. Unfortunately I then get an error "manager.vdevice.drivers.zwavebasic.pair.invalid_capability" for MT2647 during inclusion and a warning "invalid_commandclass..." for MT2646 after inclusion.

Sorry for commenting like that I don't know much about github.

**Comment by bramoosterhuis** [comment]
No problem, Did you use the alpha branch for testing? Because I fixed the
productTypeId's for the devices on that branch.

see:
https://github.com/bramoosterhuis/com.devolo/tree/alpha

thanks for testing



Met vriendelijke groet

Bram Oosterhuis
www.bramoosterhuis.com

2017-03-09 17:22 GMT+01:00 sporca <notifications@github.com>:

**Comment by sporca** [comment]
Yes, last tests were with alpha branch and very promising. productTypeIDs are correct and inclusion for motion and door sensors work. I changed "id": 22, "size": 1, to "value": **0** because the sensors started spamming each change of lux.
As said in post before the metering plug and remote keyfob are not working at the moment.

**Comment by bramoosterhuis** [comment]
I set parameter 22 and 21 default to 0  (disable differential reports) now and added a checkbox to the config so you can enable it if required. The remote key fob is not working due to some issue with homey for details see: https://github.com/athombv/homey/issues/1261. But I also own a key fob (MT2653) so that will be tested with the next release of homey's firmware. I will look into the invalid_commandClass

**Comment by bramoosterhuis** [comment]
@sporca I corrected the command classes and added the settings for the wall plugs. Let me know if this fixes the issue for the MT2646.

**Comment by sporca** [comment]
Your amazing. First test with 1 motion sensor, 1 wallplug and 1 door sensor is working great.
Some descriptions of settings in wallplug (MT2646) are not 100% correct (e.g. switch off auto...).
I know add more devices and test during weekend.

BTW. do you have a paypal donation link. I'd like to buy you some beer. :-)

**Comment by Barbados02** [comment]
Thank you for your work, just installed the alpha and my door contact is working.
Will you push this version to the app store or are more changes on the way?

Thanks again!

**Comment by sporca** [comment]
I'm still testing. Now with 3 Door-Sensors, 7 Motion Sensors and 3 Wall Plugs.

So far it's working good. I changed PIR-Sensitivity to 99, otherwise it's not detecting properly. Sometimes Motion Detection is very slow or not working reliably.
I think this is a problem with Homey Firmware.

Sometimes getting Transmit Error, but think this is Homey Firmware problem, too.
_ProcessSendData[9903]: Error: TRANSMIT_COMPLETE_NO_ACK_

One strange issue I have is with only one wall plug that is spamming his metering data every 30 seconds.
_Received application command for COMMAND_CLASS_METER, data:_

Pushing higher Watt Meter Reporting Period to the device does not change the behaviour. Maybe I have to re-include this plug, because it's working with the other 2.

I'm very happy with your work, absolutely cool!

**Comment by bramoosterhuis** [comment]
Good to read that every think seems to work smooth again. I submitted v1.0.3 of the app to a beta release. It will become available for everyone as soon as Athom approves it.
